File: 算法/leetcode/1748.唯一元素的和.cpp
/*
 * @lc app=leetcode.cn id=1748 lang=cpp
 *
 * [1748] 唯一元素的和
 */
#include <iostream>
#include <map>
#include <vector>
using namespace std;
// @lc code=start
class Solution {
 public:
  int sumOfUnique(vector<int>& nums) {
    map<int, int> mymap;
    for (int i = 0; i < nums.size(); i++) {
      mymap.insert(pair<int, int>(i,1));
      std::cout<<mymap[i]<<endl;
    }

    int sum = 0;

    for (auto i = mymap.begin(); i != mymap.end(); ++i) {
      std::cout<<i->first<<endl;
    }
    return sum;
  }
};

int main() {
  vector<int> nums = {1, 2, 3, 2};
  Solution s;
  s.sumOfUnique(nums);
  return 0;
}
// @lc code=end

File: 算法/leetcode/1748.唯一元素的和.js
/*
 * @lc app=leetcode.cn id=1748 lang=javascript
 *
 * [1748] 唯一元素的和
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var sumOfUnique = function (nums) {
  let hash = {};
  let sum=0;
  for (let i = 0; i < nums.length; i++) {
    hash[nums[i]] = hash[nums[i]] + 1 || 1;
  }
  for (const key in hash) {
    if (hash[key] === 1) {
      sum += parseInt(key);
    }
  }
  return sum;
};
// @lc code=end

console.log(sumOfUnique([1, 2, 3, 2]));
